src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -257,11 +260,22 @@ bool java_bytecode_languaget::typecheck(
     get_message_handler());
 
   // now typecheck all
-  if(java_bytecode_typecheck(
-       symbol_table, get_message_handler(), string_refinement_enabled))
-    return true;
+  auto res = java_bytecode_typecheck(
+       symbol_table, get_message_handler(), string_refinement_enabled);
 
-  return false;
+  generate_java_generic_typet new_symbol_generator(symbol_table, get_message_handler());
+  // check out the symbols in the symbol table at this point to see if we
+  // have a a generic type in.
+  for(const auto &symbol : symbol_table.symbols)
+  {
+    if(is_java_generic_type(symbol.second.type))
+    {
+      auto symb=new_symbol_generator(to_java_generic_type(symbol.second.type));
```

Comment:
Doesn't the new_symbol_generator add it to the symbol table too? Perhaps it shouldn't? But I think you're adding it twice currently.

Reply:
Nit: use whole words for variable names, e.g. `new_class_symbol`

Reply:
Yes, you are right. Due to the design of `generate_class_stub`, which is called by `new_symbol_generator`, it gets added two times. I removed this one.